packages/azure-http-specs/specs/azure/resource-manager/resources/mockapi.ts

```diff
@@ -240,115 +239,59 @@ Scenarios.Azure_ResourceManager_Resources_ExtensionsResources_createOrUpdate = p
     },
     response: {
       status: 200,
-      body: json(validResourceGroupExtensionsResource),
+      body: json(validResource),
     },
     kind: "MockApiDefinition",
-  },
-  {
-    uri: "/subscriptions/:subscriptionId/providers/Azure.ResourceManager.Resources/extensionsResources/:extensionName",
-    method: "put",
-    request: {
-      params: {
-        subscriptionId: SUBSCRIPTION_ID_EXPECTED,
-        extensionName: "extension",
-        "api-version": "2023-12-01-preview",
-      },
-      body: {
-        properties: {
-          description: "valid",
-        },
-      },
-    },
-    response: {
-      status: 200,
-      body: json(validSubscriptionExtensionsResource),
-    },
-    kind: "MockApiDefinition",
-  },
-  {
-    uri: "/providers/Azure.ResourceManager.Resources/extensionsResources/:extensionName",
-    method: "put",
-    request: {
-      params: {
-        extensionName: "extension",
-        "api-version": "2023-12-01-preview",
-      },
-      body: {
-        properties: {
-          description: "valid",
-        },
-      },
-    },
-    response: {
-      status: 200,
-      body: json(validTenantExtensionsResource),
-    },
-    kind: "MockApiDefinition",
-  },
-  {
-    uri: "/subscriptions/:subscriptionId/resourceGroups/:resourceGroup/providers/Azure.ResourceManager.Resources/topLevelTrackedResources/:topLevelResourceName/providers/Azure.ResourceManager.Resources/extensionsResources/:extensionName",
-    method: "put",
-    request: {
-      params: {
-        subscriptionId: SUBSCRIPTION_ID_EXPECTED,
-        resourceGroup: RESOURCE_GROUP_EXPECTED,
-        topLevelResourceName: "top",
-        extensionName: "extension",
-        "api-version": "2023-12-01-preview",
-      },
-      body: {
-        properties: {
-          description: "valid",
-        },
-      },
-    },
-    response: {
-      status: 200,
-      body: json(validResourceExtensionsResource),
-    },
-    kind: "MockApiDefinition",
-  },
+  };
+}
+Scenarios.Azure_ResourceManager_Resources_ExtensionsResources_createOrUpdate = passOnSuccess([
+  createExtensionCreateMockApiDefinition(
+    RESOURCE_GROUP_SCOPE_URI,
+    EXTENSION_RESOURCE_NAME,
+    validResourceGroupExtensionsResource,
+  ),
+  createExtensionCreateMockApiDefinition(
+    SPLIT_RESOURCE_GROUP_SCOPE_URI,
+    EXTENSION_RESOURCE_NAME,
+    validResourceGroupExtensionsResource,
+  ),
```

Comment:
I suspect this kind of definition would require client to write test for **BOTH** RESOURCE_GROUP_SCOPE_URI and SPLIT_RESOURCE_GROUP_SCOPE_URI. 
We only need client to write **EITHER** one of them to pass the test.

Reply:
I think both resource URIs of the formats `RESOURCE_GROUP_SCOPE_URI` and `SPLIT_RESOURCE_GROUP_SCOPE_URI` should be included in the test scope.

Reply:
Yes, for mockapi, we need to support both endpoints.
Though for clients that calls API, they only need to call either one(either  `/` or `//`). 
Like the test you write here: https://github.com/microsoft/typespec/pull/5929/files
e.g. We only need to call with resourceUri `EXTENSION_RESOURCE_SUBSCRIPTION_URL`, no need for `SPLIT_EXTENSION_RESOURCE_SUBSCRIPTION_URL`.

The `passOnSuccess` in mockapi may require all the scenarios be called successfully by client to pass this test, which is not what we want.

Reply:
Would you try [`withServiceKeys`](https://github.com/Azure/typespec-azure/blob/main/packages/azure-http-specs/specs/azure/payload/pageable/mockapi.ts#L38)?

You'll need:
1. Define 4 serviceKeys, `Tenant`, `Subscription`, `ResourceGroup`, `Resource`.
2. Use custom handler
3. In customer hander, return same `pass: {key}` for the same scope(either `/` or `//`), e.g. `pass: Tenant` for tenant level resource.